python/nano/src/bigdl/nano/pytorch/inference/optimizer.py

```diff
@@ -285,9 +295,22 @@ def func_test(model, input_sample):
                     if precision == "fp32" and method != "original":
                         result_map[method]["accuracy"] = "not recomputed"
                     else:
-                        result_map[method]["accuracy"] =\
-                            _accuracy_calculate_helper(acce_model,
-                                                       metric, validation_data)
+                        if method == "original":
+                            # test whether metric works
+                            try:
+                                result_map[method]["accuracy"] =\
+                                    _accuracy_calculate_helper(acce_model, metric,
+                                                               validation_data)
+                            except Exception:
+                                self._calculate_accuracy = False
+                                invalidInputError(False,
+                                                  "Your metric is incompatible with "
```

Comment:
explain what's the expected metric pattern in the error message

Reply:
done